ls1intum/Artemis
# `Plagiarism checks`: Add a new verdict for no plagiarism case

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


Instructors may decide that there is no plagiarism in case the the student has a good justification in the reply.

### Description

We have a new verdict `NO_PLAGIARISM`. If the instructor decides like that, there are no consequences for the student at all and the Plagiarism Case Button in the client should not be shown in anymore. Instead, there should be a button at the bottom of the page stating “Resolved plagiarism case” where the student can still navigate to the details and see the conversation.

Also, we added a modal prompt when changing Plagiarism Case from "Confirmed" to "Denied" because this is a destructive operation.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Exercise with a potential Plagiarism Case

1. Log in to Artemis
2. Navigate to Course Administration
3. Make two similar submissions with both students.
4. As the instructor, set the due date of the exercise to now and run plagiarism check from Exercise Detail page > Check Plagiarism page.
5. Notify a student about the plagiarism.
6. With the notified student, check if the exercise detail page shows the red `Plagiarism Case` button.
7. As the instructor, set the verdict as "No Plagiarism".
8. With the notified student, check if the exercise detail page does not show the red `Plagiarism Case` button but shows an info button at the bottom of the page (see screenshot) with text "Resolved Plagiarism Case".

Testing the modal to change plagiarism case from "Confirmed" to "Denied":
1. Go to a confirmed plagiarism case at Exercise Detail page > Check Plagiarism page.
2. Click "Deny".
3. You should see a modal, accepting it should change the plagiarism case from "Confirmed" to "Denied".

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| plagiarism-case-instructor-detail-view.component.ts | 79.54% | 95.23% |
| plagiarism-cases.service.ts | 71.42% | 100% |
| plagiarism-case-verdict.component.ts | 85.71% | 92.3% |
| plagiarism-header.component.ts | 80% | 100% |
| course-exercise-details.component.ts | 58.47% | 76.92% |
| PlagiarismCaseResource.java | 75% | 93% |

### Screenshots



![image](https://user-images.githubusercontent.com/18427209/182380822-9b9d2d10-658b-419d-a96e-57bd3896aa5c.png)

![image](https://user-images.githubusercontent.com/18427209/182366377-eae09fbd-d820-41f3-93f8-3066555c1daa.png)

![image](https://user-images.githubusercontent.com/18427209/182494995-649f4845-50b0-47cf-a635-ecc4bcd6bf56.png)
